libraries/AP_Mount/AP_Mount_STorM32_native.cpp

```diff
@@ -0,0 +1,497 @@
+#include "AP_Mount_STorM32_native.h"
+
+extern const AP_HAL::HAL& hal;
+
+AP_Mount_STorM32_native::AP_Mount_STorM32_native(AP_Mount &frontend, AP_Mount::mount_state &state, uint8_t instance) :
+    AP_Mount_Backend(frontend, state, instance),
+    _initialised(false),
+    _armed(false),
+    _send_armeddisarmed(false)
+{
+    _uart = nullptr;
+    _mount_type = AP_Mount::Mount_Type_None; //the mount type will be determined in init()
+
+    _task_time_last = 0;
+    _task_counter = TASK_SLOT0;
+
+    _bitmask = SEND_STORM32LINK_V2 | SEND_CMD_SETINPUTS | SEND_CMD_DOCAMERA;
+
+    _status.pitch_deg = _status.roll_deg = _status.yaw_deg = 0.0f;
+    _status_updated = false;
+
+    _target_to_send = false;
+    _target_mode_last = MAV_MOUNT_MODE_RETRACT;
+}
+
+//------------------------------------------------------
+// AP_Mount_STorM32_native interface functions
+//------------------------------------------------------
+
+// init - performs any required initialisation for this instance
+void AP_Mount_STorM32_native::init(const AP_SerialManager& serial_manager)
+{
+    //from instance we can determine its type, we keep it here since that's easier/shorter
+    _mount_type = _frontend.get_mount_type(_instance);
+
+    // it should never happen that it's not one of them, but let's enforce it, to not depend on the outside
+    if (_mount_type != AP_Mount::Mount_Type_STorM32_native) {
```

Comment:
_mount_type serves two purposes:
* the same code can be used for various serial data streams, with little modifications
examples would be UART and CAN
* I "missuse" _mount_type as a state variable. This can be avoided, but would need adding then an additional state variable ... I found that a bit too much, so I went this way

Reply:
> the same code can be used for various serial data streams, with little modifications
examples would be UART and CAN

Ok, that's a reasonable use.

> I "missuse" _mount_type as a state variable. This can be avoided, but would need adding then an additional state variable ... I found that a bit too much, so I went this way

I don't see why you need an additional state variable. And you definitely shouldn't use this as a state variable.